odl/deform/linear_deformation.py

```diff
@@ -0,0 +1,192 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Shape-based reconstruction using optimal information transportation.
+
+The Fisher-Rao metric is used in regularization term. And L2 data matching
+term is used in fitting term.
+"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+import numpy as np
+
+from odl.operator import Operator
+from odl.discr import DiscreteLp, Divergence
+from odl.space import ProductSpace
+
+__all__ = ('LinearDeformation', 'MassPreservingLinearDeformation')
+
+
+class LinearDeformation(Operator):
```

Comment:
That "somehow" would be interesting to learn more about :-)
No, honestly, for a fixed deformation, one could actually have something diagnostics-like that does a standardized test (if there is such a thing for diffeos). For variable deformation, it's harder.
On the other hand, I guess once we don't have a diffeo any longer, things just break down in the worst case. We can decide to just happily go on until the code crashes or produces NaNs.

Regarding the name, I didn't want to write `LinearizedFixedDeformationDeformation`, `Diffeo` was just nicer and shorter. Shouldn't imply anything.

As far as the total length is concerned, I'm not so worried. We probably have longer stuff, and this is still not quite STIR level. We could try to crank down on the `Linearized` part, though. Alternatives?